pyvast/pyvast.py

```diff
@@ -0,0 +1,48 @@
+"""python vast module
+
+"""
+
+import asyncio
+import logging
+import pyarrow
+import sys
+
+def str_escape(x):
+    return '"' + x.replace('"', '\\"') + '"'
+
+
+async def spawn(*args):
+    """Spawns a process asynchronously."""
+    proc = await asyncio.create_subprocess_exec(
+        *args, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+    )
+    stdout, stderr = await proc.communicate()
+    logger = logging.getLogger("threat-bus")
+    logger.debug(stderr.decode().strip())
+    return stdout, proc.returncode
+
+
+class VAST:
+    """A VAST node handle"""
+
+    def __init__(self, app='vast', endpoint='localhost:42000'):
+        self.logger = logging.getLogger("vast")
+        self.app = app
+        self.endpoint = endpoint
+        self.logger.debug("connecting to vast on %s", self.endpoint)
+
+    async def select(self, expression):
+        """Extracts data from VAST according to a query"""
+        self.logger.debug("running query %s", expression)
+        raw = await spawn(self.app, "export", "arrow", expression)
+        istream = pyarrow.input_stream(raw)
+        reader = pyarrow.ipc.RecordBatchStreamReader(istream)
+        table = reader.read_all()
+        return table
+
+    async def try_connect(self):
```

Comment:
```suggestion
    async def status(self):
```

I would make this function return the status as JSON or `None` if it's impossible to connect.